crates/board/src/crypto.rs

```diff
@@ -158,8 +175,57 @@ impl crate::Supported for sha2::Sha256 {}
 impl crate::Supported for sha2::Sha384 {}
 
 #[cfg(feature = "internal-software-crypto-hmac")]
-impl<D: Support<bool> + Default + BlockSizeUser + Update + FixedOutput + HashMarker> Support<bool>
-    for hmac::SimpleHmac<D>
+impl<D: Support<bool> + Default + BlockSizeUser + Update + FixedOutputReset + HashMarker>
+    Support<bool> for hmac::SimpleHmac<D>
 {
     const SUPPORT: bool = D::SUPPORT;
 }
+
+/// Extends APIs with infallible operations to support errors.
+///
+/// Infallible operations may return incorrect values. Those results must not be used until
+/// `last_error()` is called and returns `Ok(())`, signifying that the return value is correct.
+#[cfg(any(feature = "internal-api-crypto-hash", feature = "internal-api-crypto-hmac"))]
+pub trait LastError {
+    /// Returns the last error of infallible operations.
+    fn last_error(&self) -> Result<(), Error>;
+}
+
+// Wrapper API around Hash that calls last_error().
+#[cfg(feature = "internal-api-crypto-hash")]
+#[allow(missing_docs)]
+pub struct HashApi<T: Hash>(T);
+
+// Wrapper API around Hmac that calls last_error().
+#[cfg(feature = "internal-api-crypto-hmac")]
+#[allow(missing_docs)]
+pub struct HmacApi<T: Hmac>(T);
+
+#[allow(missing_docs)]
+#[cfg(feature = "internal-api-crypto-hash")]
+impl<T: Hash> HashApi<T> {
+    // Constructor of HashApi
+    pub fn new() -> Result<Self, Error> {
+        let hash = T::default();
+        let error = T::last_error(&hash);
```

Comment:
nit: it should be possible to call last_error as a method
```suggestion
        let error = hash.last_error();
```